Grammar and typo fixes

viliam-durina opened the pull request:
A random collection while reading through the code.

viliam-durina committed:
Grammar and typo fixes

A random collection while reading through the code.

viliam-durina committed:
Even more grammar fixes

shubhamvishu commented on lucene/core/src/java/org/apache/lucene/index/DocumentsWriterPerThreadPool.java:
```diff
@@ -184,7 +184,7 @@ List<DocumentsWriterPerThread> filterAndLock(Predicate<DocumentsWriterPerThread>
    * Removes the given DWPT from the pool unless it's already been removed before.
    *
-   * @return <code>true</code> iff the given DWPT has been removed. Otherwise <code>false</code>
+   * @return <code>true</code>, iff the given DWPT has been removed. Otherwise <code>false</code>
```
Maybe fix /iff/if/ since you are at it?

viliam-durina replied:
"Iff" is correct, it means "if and only if", see https://www.merriam-webster.com/dictionary/iff

msokolov replied:
Since this is all about syntax, grammar, etc, we might as well italicize _iff_ since that is how it is generally done, as in the example you cited.

msokolov replied:
also, the comma you inserted is not correct (try replacing iff with "if and only if" - we wouldn't use a comma before "if").

msokolov commented on lucene/core/src/java/org/apache/lucene/index/IndexCommit.java:
```diff
@@ -25,9 +25,9 @@
- * <p>Changes to the content of an index are made visible only after the writer who made that change
- * commits by writing a new segments file (<code>segments_N</code>). This point in time, when the
- * action of writing of a new segments file to the directory is completed, is an index commit.
+ * <p>Changes to the content of an index are made visible only after the writer which made that
```
sorry this is wrong. We could use `that`, but `who` is also correct - no need to "fix" it,

msokolov commented on lucene/core/src/java/org/apache/lucene/index/IndexCommit.java:
```diff
@@ -25,9 +25,9 @@
- * commits by writing a new segments file (<code>segments_N</code>). This point in time, when the
- * action of writing of a new segments file to the directory is completed, is an index commit.
+ * <p>Changes to the content of an index are made visible only after the writer which made that
+ * change commits by writing a new segments file (<code>segments_N</code>). This point in time, when
```
OK this sentence is awkward, but what you've done here doesn't really address the core problem. I think it really would be most correct as:  "_The_ point in time (no comma) when ... is completed (no comma) is an index commit".

msokolov commented on lucene/core/src/java/org/apache/lucene/index/IndexOptions.java:
```diff
@@ -33,7 +33,7 @@ public enum IndexOptions {
   /**
    * Only documents and term frequencies are indexed: positions are omitted. This enables normal
-   * scoring, except Phrase and other positional queries will throw an exception.
+   * scoring, except phrase and other positional queries will throw an exception.
```
I think I would prefer to avoid this ambiiguity by using  `PhraseQuery` explicitly

msokolov commented on lucene/core/src/java/org/apache/lucene/index/IndexWriter.java:
```diff
@@ -101,8 +101,8 @@
  * continue to search the "point in time" snapshot they had opened, and won't see the newly created
- * index until they re-open. If {@link OpenMode#CREATE_OR_APPEND} is used IndexWriter will create a
- * new index if there is not already an index at the provided path and otherwise open the existing
+ * index until they re-open. If {@link OpenMode#CREATE_OR_APPEND} is used, IndexWriter will create a
```
Again, not a great sentence, but fixing it requires a bigger rewrite; I don't think that this change is better (the "will" has no subject).

msokolov commented on lucene/core/src/java/org/apache/lucene/index/IndexWriter.java:
```diff
@@ -123,13 +123,13 @@
  * IndexWriterConfig#DEFAULT_RAM_BUFFER_SIZE_MB} MB. For best indexing speed you should flush by RAM
- * usage with a large RAM buffer. In contrast to the other flush options {@link
- * IndexWriterConfig#setRAMBufferSizeMB} and {@link IndexWriterConfig#setMaxBufferedDocs(int)},
+ * usage with a large RAM buffer. In contrast to the other flush options, in case of {@link
```
let's leave this as it was. It was OK.

msokolov reviewed (commented):
I started reviewing these one by one, but as I go I find that many of these changes insert unneeded commas and other agrammatical constructions that are sometimes worse than the original (which admittedly could stand to be improved). If you really want to work on this, I am happy to give out more grammar pedantry (it tickles some kind of weird obsession I have), but maybe it is not worth our time?

viliam-durina committed:
More grammar fixes

viliam-durina committed:
Revert some changes

viliam-durina committed:
Run gradlew tidy

viliam-durina commented:
Thanks for the review :+1: 

I reverted all the disputed changes, agree we should not go into unproductive discussions. I added a few new ones, they are in a separate commit to simplify the review.

I'll only have note to the `writer ... who` (https://github.com/apache/lucene/pull/14019#discussion_r1864958353) - "who" should be used only for persons, this writer is not a person. But I reverted it anyway.

msokolov commented on lucene/core/src/java/org/apache/lucene/index/IndexWriter.java:
```diff
@@ -141,7 +141,7 @@
  * "point in time" commits alive in the index for some time, either because this is useful for your
  * application, or to give readers enough time to refresh to the new commit without having the old
- * commit deleted out from under them. The latter is necessary when multiple computers take turns
+ * commit deleted under their hands. The latter is necessary when multiple computers take turns
```
this sounds like an idiom of sort, but I've never heard it -- I think the existing phrasing is more idiomatic of the English I know anyway.

msokolov commented on lucene/core/src/java/org/apache/lucene/index/IndexWriterConfig.java:
```diff
@@ -509,28 +509,28 @@ public IndexWriterConfig setCheckPendingFlushUpdate(boolean checkPendingFlushOnU
+   * example, to keep all document modifications for a certain time interval or the last N
+   * operations if some kind of sequence ID is available in the index.
    *
    * <p>Currently there is no API support to un-delete a soft-deleted document. In oder to un-delete
```
Can we fix "In oder" to be "In order"?

msokolov reviewed (changes requested):
Thank you for all the fixes! I reviewed them all this time, and I think there are just a couple of changes I'd like to see before merging.

viliam-durina committed:
Address review comments

msokolov reviewed (approved):
Looks good, thank you!

msokolov commented:
BTW I am fascinated by your github avatar pic! I feel like I have seen it before but not sure where? I tried image search but it does not find it. Is it something well known?

viliam-durina commented:
> BTW I am fascinated by your github avatar pic! I feel like I have seen it before but not sure where? I tried image search but it does not find it. Is it something well known?

It's an illustration from a children book by Albín Brunovský :smile:

msokolov commented:
wow those books would scare any child that I know!
